src/Application/Migrations/__2019-09-29_skip_dependent.sql

```diff
@@ -0,0 +1,184 @@
+BEGIN;
+	
+SET ROLE TO postgres;
+
+ALTER TABLE tbl_project_ticket_state
+  ADD COLUMN skip_on_dependent boolean NOT NULL DEFAULT false;
+
+ALTER TABLE tbl_ticket_state
+  ADD COLUMN skippable_on_dependent boolean NOT NULL DEFAULT false;
+
+
+UPDATE tbl_ticket_state
+SET skippable_on_dependent = true
+WHERE ticket_type = 'encoding'
+  AND ticket_state IN ('checking', 'checked', 'postprocessing', 'postprocessed', 'ready to release');
+
+DROP FUNCTION IF EXISTS ticket_state_next(bigint);
+DROP FUNCTION IF EXISTS ticket_state_next(bigint, enum_ticket_type);
+DROP FUNCTION IF EXISTS ticket_state_next(bigint, enum_ticket_type, enum_ticket_state);
+
+DROP FUNCTION IF EXISTS ticket_state_previous(bigint);
+DROP FUNCTION IF EXISTS ticket_state_previous(bigint, enum_ticket_type);
+DROP FUNCTION IF EXISTS ticket_state_previous(bigint, enum_ticket_type, enum_ticket_state);
+
+DROP FUNCTION IF EXISTS ticket_state_commence(bigint, enum_ticket_type);
+DROP FUNCTION IF EXISTS ticket_state_commence(bigint);
+
+
+CREATE OR REPLACE FUNCTION update_ticket_next_state()
+  RETURNS trigger AS
+$BODY$
+  DECLARE
+   next_state record;
+  BEGIN
+    next_state := ticket_state_next(NEW.id, NEW.ticket_state);
+
+    NEW.ticket_state_next := next_state.ticket_state;
+    NEW.service_executable := next_state.service_executable;
+
+    RETURN NEW;
+  END
+$BODY$
+LANGUAGE plpgsql VOLATILE;
+
+
+CREATE OR REPLACE FUNCTION update_all_tickets_progress_and_next_state(param_project_id bigint)
+  RETURNS VOID AS
+$BODY$
+  BEGIN
+
+    UPDATE tbl_ticket t SET
+      (progress, ticket_state_next, service_executable)
+        = (tp, (n).ticket_state, (n).service_executable)
+    FROM (
+      SELECT id, ticket_state_next(t2.id) AS n, ticket_progress(t2.id) as tp
+      FROM tbl_ticket t2
+      WHERE t2.project_id = param_project_id AND param_project_id IS NOT NULL
+    ) AS x
+    WHERE t.id = x.id;
```

Comment:
Not sure why that needs a subquery?!

Reply:
The function "ticket_state_next" shall not be called twice per row. No idea how to use the result of a function call in a SET block of an UPDATE statement, since you need the state and its executable flag separately.

Reply:
I see